0000-windows-verbatim-paths.md

```diff
@@ -0,0 +1,79 @@
+- Feature Name: windows_verbatim_paths
+- Start Date: 2015-11-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Transparently use the long pathname API in Windows, allowing Rust programs to
+deal with paths longer than 260 characters without having to implement their
+own pathname handling.
+
+# Motivation
+[motivation]: #motivation
+
+Windows has multiple ways to handle paths:
+
+1. ANSI paths, as in `CreateFileA`, restricted to one code page with 256
+   characters and paths of 260 characters: `r"C:\"`. Rust doesn't use these.
+2. Traditional Unicode paths, as in `CreateFileW`, accepts a superset of
+   UTF-16, as it allows for unmatched surrogates, maximum of 260 UTF-16 code
+   units: `r"C:\"`. Rust currently uses these.
+3. Extended Unicode paths, also accepted by `CreateFileW`, accepts the same
+   superset of UTF-16 but allows for roughly 32768 code units.
+
+Rust currently uses the second path style, unless the application explicitly
+chooses to implement the path handling needed for 3. Since paths of the other
+forms can be converted into the third form, Rust could implement this in the
+standard library. Rust applications would consistently and robustly support
+long path names.
+
+# Detailed design
+[design]: #detailed-design
+
+The path translation will happen directly above the WinAPI layer, when the
+`OsString`s are converted to arrays of 16-bit integers (UTF-16 code units).
+
+- `C:\foo\bar` to `\\?\C:\foo\bar`.
+- `\foo\bar` to the current working directory (CWD) joined with `\foo\bar`.
+- `foo\bar` to CWD joined with `foo\bar`.
+- `C:foo\bar`. This depends if on the current working directory. If it is on
```

Comment:
I'd like to point out that each drive does store its own current working directory. I tested this before, so we cannot assume the non-current drive's current working directory is the root.